packages/vitest-plugin-storybook/src/test-utils.ts

```diff
@@ -0,0 +1,59 @@
+export { setViewport } from './viewports'
+import type { TaskContext, Task, TaskMeta } from 'vitest'
+import type { UserOptions } from './types'
+import { setViewport } from './viewports'
+
+type TagsFilter = Required<UserOptions['tags']>
+
+export const shouldSkip = (storyTags: string[], tagsFilter: TagsFilter) => {
+  return tagsFilter?.skip.some((tag) => storyTags.includes(tag))
+}
+
+export const shouldRun = (storyTags: string[], tagsFilter: TagsFilter) => {
+  const isIncluded =
+    tagsFilter?.include.length === 0 ||
+    tagsFilter?.include.some((tag) => storyTags.includes(tag))
+  const isNotExcluded = tagsFilter?.exclude.every(
+    (tag) => !storyTags.includes(tag)
+  )
+
+  return isIncluded && isNotExcluded
+}
+
+export const testStory = (
+  exportName: string,
+  modulePath: string,
+  composeStoriesFn: ComposeStoriesFn,
+  tagsFilter: TagsFilter
+) => {
+  return async ({ task, skip }: TaskContext) => {
+    const Story = (await getStories(modulePath, composeStoriesFn))[exportName]
+    if (Story === undefined || shouldSkip(Story.tags, tagsFilter)) {
+      skip()
+    }
+
+    const _task = task as Task & {
+      meta: TaskMeta & { storyId: string; hasPlayFunction: boolean }
+    }
+    _task.meta.storyId = Story.id
+    _task.meta.hasPlayFunction = !!Story.play
+    await setViewport(Story.parameters.viewport)
```

Comment:
Sure thing, but for now we can't do that as it would require substantial changes to the viewport addon right. In the meantime this helps us get compatibility with stories that depend on viewport size

Reply:
We could do this right?

```tsx
// code/addons/viewport/src/preview.ts
import type { BeforeEach } from '@storybook/csf';

export const initialGlobals = { viewport: 'reset', viewportRotated: false };

export const beforeEach: BeforeEach = async ({ parameters }) => {
  if (import.meta.vitest) {
    await setViewport(parameters.viewport);
  }
};
```

But I agree, let's think about that design for another PR